### PR TITLE
INTMDB-422 encryption at rest

### PR DESCRIPTION
**Pull request by rdharmedran**
**## Description**
             Feature update: encryption-at-rest 

**Link to any related issue(s):** 
                                    https://jira.mongodb.org/browse/INTMDB-422
**Type of change:**
[✅] Bug fix (non-breaking change which fixes an issue)
[✅ ] New feature (non-breaking change which adds functionality)
 Breaking change (fix or feature that would cause existing functionality to not work as expected)
[✅ ] This change requires a documentation update

**Required Checklist:**
[✅] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
 [✅] I have added tests that prove my fix is effective or that my feature works
[✅]  I have added any necessary documentation (if appropriate)
[✅]  I have run make fmt and formatted my code

**Summary**

- Added configuration for CFN TEST in test folder

- Replaced IAM user related parameters with IAM Role ID (in model class) as per latest documentation.Please refer this link for more details https://www.mongodb.com/docs/atlas/security-aws-kms/.

- Read function returned already deleted configuration.So we have added condition to check if exist.

- Delete function was not checking if configuration is already exist before delete it.We have added condition for the same

- We have added enough log statements in the code

### CODE REVIEW

**Comment by Zuhairahmed** [comment]
@rdharmedran can we add GitHub checks to this PR (build + linting)?

File: cfn-resources/encryption-at-rest/README.md [hunk 1]
```diff
@@ -2,17 +2,65 @@
...
-    $ make
-    $ cfn submit
-    ...
\ No newline at end of file
+TAGS=logging make
+cfn submit --verbose --set-default
+
+## Cloudformation Examples
+
+Please use the [CFN Template](test/encryptionatrest.sample-template.yaml)
```
Comment (Zuhairahmed) [inline cfn-resources/encryption-at-rest/README.md:40]:
```suggestion
Please see the [CFN Template](test/encryptionatrest.sample-template.yaml) for example resource 
```

File: cfn-resources/encryption-at-rest/README.md [hunk 2]
```diff
@@ -2,17 +2,65 @@
...
+
+## Integration Testing w/ AWS
+
+The [/quickstart-mongodb-atlas/scripts/launch-x-quickstart.sh](launch-x-quickstart.sh) script
```
Comment (Zuhairahmed) [inline cfn-resources/encryption-at-rest/README.md:44]: This link doesn't work, can you double check to bring user to correct GitHub page

File: cfn-resources/encryption-at-rest/README.md [hunk 3]
```diff
@@ -2,17 +2,65 @@
...
+And then you can create the stack with a helper script it insert the apikeys for you:
+
+
+```bash
+repo_root=$(git rev-parse --show-toplevel)
+source <(${repo_root}/quickstart-mongodb-atlas/scripts/export-mongocli-config.py)
+${repo_root}/quickstart-mongodb-atlas/scripts/launch-x-quickstart.sh ${repo_root}/cfn-resources/encryption-at-rest/test/encryptionatrest.sample-template.yaml SampleAccessList1 ParameterKey=ProjectId,ParameterValue=<YOUR_PROJECT_ID> ParameterKey=CustomerMasterKeyID,ParameterValue=<CustomerMasterKeyID> ParameterKey=RoleID,ParameterValue=<RoleID> ParameterKey=Region,ParameterValue=<Region> ParameterKey=Enabled,ParameterValue=<true or false>
+```
+
+For more information see: MongoDB Atlas API Cloud BackUp [Endpoint](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Encryption-at-Rest-using-Customer-Key-Management) Documentation.
```
Comment (Zuhairahmed) [inline cfn-resources/encryption-at-rest/README.md:66]:
```suggestion
For more information see: MongoDB Atlas API [Cloud BackUp Endpoint](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Encryption-at-Rest-using-Customer-Key-Management) Documentation.
```

File: cfn-resources/encryption-at-rest/README.md [hunk 4]
```diff
@@ -2,17 +2,65 @@
 
 ## Description
 This resource allows administrators to enable, disable, configure, and retrieve the configuration for Encryption at Rest.
```
Comment (Zuhairahmed) [inline cfn-resources/encryption-at-rest/README.md:4]:
```suggestion
Returns and edits the Encryption at Rest using Customer Key Management configuration
```
can we make sure to leverage the Open API docs as much as possible for these definitions? https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Encryption-at-Rest-using-Customer-Key-Management

File: cfn-resources/encryption-at-rest/docs/README.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
+# MongoDB::Atlas::EncryptionAtRest
+
+This resource allows administrators to enable, disable, configure, and retrieve the configuration for Encryption at Rest.
```
Comment (Zuhairahmed) [inline cfn-resources/encryption-at-rest/docs/README.md:3]: Update definition here to match with other readme file

**Review by Zuhairahmed — COMMENTED** [review]
Few doc comments below

File: cfn-resources/encryption-at-rest/README.md (hunk 1 above)
Comment (rdharmedran) [reply cfn-resources/encryption-at-rest/README.md]: This issue is fixed

File: cfn-resources/encryption-at-rest/README.md (hunk 2 above)
Comment (rdharmedran) [reply cfn-resources/encryption-at-rest/README.md]: This issue is fixed

File: cfn-resources/encryption-at-rest/README.md (hunk 3 above)
Comment (rdharmedran) [reply cfn-resources/encryption-at-rest/README.md]: This issue is fixed

File: cfn-resources/encryption-at-rest/README.md (hunk 4 above)
Comment (rdharmedran) [reply cfn-resources/encryption-at-rest/README.md]: This issue is fixed

File: cfn-resources/encryption-at-rest/docs/README.md (hunk 5 above)
Comment (rdharmedran) [reply cfn-resources/encryption-at-rest/docs/README.md:3]: This issue is fixed

**Review by Zuhairahmed — APPROVED** [review]
LGTM! Thanks again @rdharmedran !

**Review by govardhanpagidi — APPROVED** [review]
LGTM!

**Review by swattyT — APPROVED** [review]
LGTM!

File: cfn-resources/encryption-at-rest/cmd/resource/resource.go [hunk 6]
```diff
@@ -106,3 +234,21 @@ func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.P
 		ResourceModel:   currentModel,
 	}, nil
 }
+
+// function to set logger
+func setup() {
+	util.SetupLogger("mongodb-atlas-project")
```
Comment (SuperMohit) [inline cfn-resources/encryption-at-rest/cmd/resource/resource.go:240]: Provide a specific resource name.

**Pull request merged**